### PR TITLE
fix formatter regex

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: CEL formatter support call function
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] #https://github.com/envoyproxy/envoy/issues/33569
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/formatter/substitution_formatter.cc [hunk 1]
```diff
@@ -38,7 +38,7 @@ const std::regex& SubstitutionFormatParser::commandWithArgsRegex() {
   //                              _________________  _______________  _____________
   //                              |               |  |             |  |           |
   CONSTRUCT_ON_FIRST_USE(std::regex,
-                         R"EOF(^%((?:[A-Z]|[0-9]|_)+)(?:\(([^\)]*)\))?(?::([0-9]+))?%)EOF");
+                         R"EOF(^%((?:[A-Z]|[0-9]|_)+)(?:\(([^\)]*|[^\}]*)\))?(?::([0-9]+))?%)EOF");
```
Comment (kyessenov) [inline source/common/formatter/substitution_formatter.cc:41]: Can you update the comment that spells out the meaning of the regex above and below the regex?

File: test/extensions/formatter/cel/cel_test.cc [hunk 2]
```diff
@@ -130,6 +130,22 @@ TEST_F(CELFormatterTest, TestMaxLength) {
   EXPECT_EQ("/original", formatter->formatWithContext(formatter_context_, stream_info_));
 }
 
+TEST_F(CELFormatterTest, TestContains) {
+  const std::string yaml = R"EOF(
+  text_format_source:
+    inline_string: "%CEL(request.url_path.contains('request'))%"
```
Comment (kyessenov) [inline test/extensions/formatter/cel/cel_test.cc:136]: I'm not sure it's actually possible to parse this well without escaping. Consider an expression `request.path.contains(')')`. A parser needs to be aware of quoting inside CEL, to properly delineate the expression. That's not doable with regex for eBNF grammar. I think you have to require escaping for `)` character?

Comment (zirain) [reply test/extensions/formatter/cel/cel_test.cc:136]: I think both `)` and `%` need escaping.

File: source/common/formatter/substitution_formatter.cc (hunk 1 above)
Comment (zirain) [reply source/common/formatter/substitution_formatter.cc:41]: update base on https://regex101.com/r/rmBzTD/1

**Comment by adisuissa** [comment]
Can you please add a release note?

Assigning Yan for a final pass after that.
/assign @yanavlasov 

/wait

**Comment by yanavlasov** [comment]
Waiting for format fixes.

/wait

**Comment by zirain** [comment]
> Waiting for format fixes.
> 
> /wait

fixed

**Comment by yanavlasov** [comment]
Thanks
@kyessenov do you have any more comments for this PR?

/wait-any

**Review by yanavlasov — APPROVED** [review]

**Comment by kyessenov** [comment]
This doesn't look like a good PR to me - at a minimum the negated character classes should be combined in one `[^%}]`?

**Pull request merged**